packages/map/src/components/EditorPanel.js

```diff
@@ -1848,6 +1854,57 @@ const EditorPanel = props => {
                       </button>
                     </fieldset>
                   )}
+                  {'category' === state.legend.type && (
+                    <fieldset className='primary-fieldset edit-block'>
+                      <label>
+                        <span className='edit-label'>
+                          Additional Category
+                          <Tooltip style={{ textTransform: 'none' }}>
+                            <Tooltip.Target>
+                              <Icon display='question' style={{ marginLeft: '0.5rem' }} />
+                            </Tooltip.Target>
+                            <Tooltip.Content>
+                              <p>You can provide additional categories to ensure they appear in the legend</p>
+                            </Tooltip.Content>
+                          </Tooltip>
+                        </span>
+                      </label>
+                      {state.legend.additionalCategories && state.legend.additionalCategories.map((val, i) => (
+                        <fieldset className='edit-block' key={val}>
+                          <button
+                            className='remove-column'
+                            onClick={event => {
```

Comment:
Nothing to update here - I just wanted to point out that in a few lines in the codebase, I've been seeing long functions on components. When they start impacting readability I've been moving them into handler functions ie `handleButtonClick`.